### ISSUE TITLE
"Error during build: .0.0 is not valid SemVer string"

### ISSUE THREAD

**Issue by ctb** [jupyterhub/repo2docker#1140]
When I try to start mybinder: https://mybinder.org/v2/gh/ngs-docs/2020-ggg-201b-rnaseq/latest?urlpath=rstudio

I get the following error message upon build:

```
Waiting for build to start...
Picked Git content provider.
Cloning into '/tmp/repo2docker9q9ck4ah'...
HEAD is now at bfa8ad2 update README to point to 201b branch
Error during build: .0.0 is not valid SemVer string
```

From random googling I'm seeing this pop up occasionally in some docker-related places, but nothing clearly connected to my issue. Any ideas?

**Comment by ctb** [comment]
it looks like a (the?) problem is that `r-base` needs to be pinned to a specific version; I had it in `binder/environment.yml` as just `r-base`, and when I set it to `r-base=4.0.5` it started building.

@raynamharris gets the credit for pointing this out to me on slack at some time in the distant past :). I'll follow up here after the build is done.

**Comment by minrk** [comment]
Exactly right, there are assumptions that self.r_version is always specified, but that isn't the case when you have `r-base` without a version pinning. The workaround is indeed to specify `r-base=4.0` or `r-base=4.0.5` etc. Should be fixed by #1141
